### DISCUSSION TITLE
SWR does not run after component initialization with useEffect([])

### DISCUSSION THREAD

**Discussion by eugeneborodkin** [vercel/swr#1219] [Q&A]
I have a functional component and an SWR fetch wrapped in a hook as follows,

```
const fetcher = url => axios.get(url).then(res => res.data);

export default function useTeleworkInfoSWR() {

    const { data, error, mutate } = useSWR('/teleworkInfo', fetcher);

    return {
        teleworkInfoSWR: data,
        teleworkInfoSWRIsLoading: !error && !data,
        teleworkInfoSWRError: error, 
        teleworkInfoSWRMutate: mutate
    }

}
```

This hook is used in a functional component which is initialized with **useEffect(.., [])** which is the standard React way of component mounting. I am noticing that SWR fetches **before** this useEffect[] initialization, and **does not** fetch or re-fetch after it:

```
        // Declare hook at the very top
        const {teleworkInfoSWR, teleworkInfoSWRIsLoading, teleworkInfoSWRError, teleworkInfoSWRMutate} = useTeleworkInfoSWR();

        // Component initialization
	useEffect(() => {
		fetchUserInfo(); // After this init fetch I need SWR to re-fetch 
	}, []);
```

So is useEffect([]) insufficient for SWR to re-validate? When exactly does SWR decide when to validate, given the above definition? What should I do to make it aware of useEffect[] re-fetching?

**Comment by shuding** [comment]
You can call `mutate()` (returned by `useSWR`) to manually trigger SWR's re-fetching.

> When exactly does SWR decide when to validate

I believe you can find this in the docs (swr.vercel.app).
